### ISSUE TITLE
Double validation when validating a field of an array nested inside a nested object

### ISSUE THREAD

**Issue by dbellavista** [Automattic/mongoose#5411]
**Do you want to request a *feature* or report a *bug*?**

**bug**

**What is the current behavior?**

When I change an array nested inside a nested object, the field validator is called twice:

1. With scope setted to the array element (correct)
2. With scope setted as the whole document (wrong)

**If the current behavior is a bug, please provide the steps to reproduce.**

You can find a minimal working example here: https://gist.github.com/dbellavista/29e1fcf66e13c19755c0fc7aab40bb57

**What is the expected behavior?**

The validator should have been called once with the subdocument as scope

**Please mention your node.js, mongoose and MongoDB version.**

* node: v8.1.2
* mongoose: 4.11.0
* mongo: v3.4.4

**Comment by vkarpov15** [comment]
Try upgrading to 4.11.1, I suspect this is a duplicate of #5363, which was fixed in that release.

**Comment by varunjayaraman** [comment]
@dbellavista can you confirm this problem persists with 4.11.1?

**Comment by dbellavista** [comment]
@varunjayaraman Still persist. The output is
```
[ { _id: 59613ece28b5cec1fd6dda24, value: true } ]
[ { _id: 59613ece28b5cec1fd6dda25, value: false },
  { _id: 59613ece28b5cec1fd6dda23, nest1: { nest2: [Object] } } ]
```

**Comment by dbellavista** [comment]
@varunjayaraman I saw the <need-repro-script> label. Doesn't suffice https://gist.github.com/dbellavista/29e1fcf66e13c19755c0fc7aab40bb57 ? I can alter it if you want

**Comment by varunjayaraman** [comment]
@dbellavista will take a look in a bit

**Comment by vkarpov15** [comment]
@dbellavista thanks for your repro script. The 'needs repro script' label means that the maintainers haven't repro-ed the issue yet, so in this case it means that we hadn't tried your repro script yet. The issue is now fixed in master and will be released with 4.11.2.
